### PR TITLE
implement master slave for both m2d and m3d

### PR DESCRIPTION
**Pull request by gmalinve**

### CODE REVIEW

**Review by Samuelopez-ansys — COMMENTED** [review]
@gmalinve Did you consider to include HFSS Master/Slave? I think it is worthy to have the same method for both... In HFSS now it is called Primary/Secondary, but it is the same concept... what do you think?

File: pyaedt/maxwell.py [hunk 1]
```diff
@@ -1552,6 +1552,162 @@ def export_element_based_harmonic_force(
...
 
+    @pyaedt_function_handler()
+    def assign_master_slave(
+        self,
+        master_entity,
+        slave_entity,
+        u_vector_origin_coordinates_master=None,
+        u_vector_pos_coordinates_master=None,
+        u_vector_origin_coordinates_slave=None,
+        u_vector_pos_coordinates_slave=None,
```
Comment (MaxJPRey) [inline pyaedt/maxwell.py:1563]: Here the default value could be `["0mm", "0mm", "0mm"]` instead of `None`.

File: pyaedt/maxwell.py [hunk 2]
```diff
@@ -1552,6 +1552,162 @@ def export_element_based_harmonic_force(
...
+                elif [x for x in u_vector_pos_coordinates_slave if not isinstance(x, str)]:
+                    raise ValueError("Elements of coordinates system must be strings in the form of ``value+unit``.")
+                elif len(u_vector_origin_coordinates_master) != 3:
+                    raise ValueError("Vector must contain 3 elements for x, y anz coordinates.")
+                elif len(u_vector_origin_coordinates_slave) != 3:
+                    raise ValueError("Vector must contain 3 elements for x, y anz coordinates.")
+                elif len(u_vector_pos_coordinates_master) != 3:
+                    raise ValueError("Vector must contain 3 elements for x, y anz coordinates.")
+                elif len(u_vector_pos_coordinates_slave) != 3:
+                    raise ValueError("Vector must contain 3 elements for x, y anz coordinates.")
```
Comment (MaxJPRey) [inline pyaedt/maxwell.py:1670]:
```suggestion
                elif len(u_vector_origin_coordinates_master) != 3:
                    raise ValueError("Vector must contain 3 elements for x, y anz coordinates.")
                elif len(u_vector_origin_coordinates_slave) != 3:
                    raise ValueError("Vector must contain 3 elements for x, y anz coordinates.")
                elif len(u_vector_pos_coordinates_master) != 3:
                    raise ValueError("Vector must contain 3 elements for x, y and z coordinates.")
                elif len(u_vector_pos_coordinates_slave) != 3:
                    raise ValueError("Vector must contain 3 elements for x, y and z coordinates.")
```

File: pyaedt/maxwell.py [hunk 3]
```diff
@@ -1552,6 +1552,162 @@ def export_element_based_harmonic_force(
...
+
+        Parameters
+        ----------
+        master_entity : int
+            ID of the master entity.
+        slave_entity : int
+            ID of the slave entity.
+        u_vector_origin_coordinates_master : list, optional
+            Master's list of U vector origin coordinates.
+            Default value is ``["0mm", "0mm", "0mm"]``.
```
Comment (MaxJPRey) [inline pyaedt/maxwell.py:1579]:
```suggestion
            The default value is ``None`` and in this case the coordinates will be ``["0mm", "0mm", "0mm"]``.
```

File: pyaedt/maxwell.py [hunk 4]
```diff
@@ -1552,6 +1552,162 @@ def export_element_based_harmonic_force(
...
+        u_vector_pos_coordinates_master : list, optional
+            Master's list of U vector position coordinates.
+            Default value is ``["0mm", "10mm", "0mm"]``.
```
Comment (MaxJPRey) [inline pyaedt/maxwell.py:1582]:
```suggestion
            The default value is ``None`` and in this case the coordinates will be ``["0mm", "0mm", "0mm"]``.
```

File: pyaedt/maxwell.py [hunk 5]
```diff
@@ -1552,6 +1552,162 @@ def export_element_based_harmonic_force(
...
+        u_vector_origin_coordinates_slave : list, optional
+            Slave's list of U vector origin coordinates.
+            Default value is ``["0mm", "0mm", "0mm"]``.
```
Comment (MaxJPRey) [inline pyaedt/maxwell.py:1585]:
```suggestion
            The default value is ``None`` and in this case the coordinates will be ``["0mm", "0mm", "0mm"]``.
```

File: pyaedt/maxwell.py [hunk 6]
```diff
@@ -1552,6 +1552,162 @@ def export_element_based_harmonic_force(
...
+        u_vector_pos_coordinates_slave : list, optional
+            Slave's list of U vector position coordinates.
+            Default value is ``["0mm", "-10mm", "0mm"]``.
```
Comment (MaxJPRey) [inline pyaedt/maxwell.py:1588]:
```suggestion
            The default value is ``None`` and in this case the coordinates will be ``["0mm", "0mm", "0mm"]``.
```

**Review by MaxJPRey — APPROVED** [review]
Great!

**Review by maxcapodi78 — APPROVED** [review]
LGTM

**Review by Samuelopez-ansys — APPROVED** [review]
LGTM

**Comment by jmaravin** [comment]
Hi, is there a way to parameterize the coordinates of the U vectors in the master and slave boundary condition? What I can see you can only input the coordinates directly through the real number...

**Pull request merged**